app/routes/goal_routes.py

```diff
@@ -1 +1,89 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, Response, request, abort, make_response
+from ..models.goal import Goal
+from ..db import db
+from ..models.task import Task
+from .route_utilities import validate_model
+
+bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+@bp.post("")
+def create_goal():
+    request_body = request.get_json()
+    new_goal = Goal.obj_from_dict(request_body)
+
+    db.session.add(new_goal)
+    db.session.commit()
+    response = {"goal": new_goal.obj_to_dict()}
+    return response, 201
+
+@bp.post("/<goal_id>/tasks")
+def add_tasks_to_goals(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+    task_ids = request_body["task_ids"]
+    tasks = Task.query.filter(Task.id.in_(task_ids)).all()
+    goal.tasks = tasks
+
+    db.session.commit()
+    response = {
+        "id": goal.id,
+        "task_ids": task_ids
```

Comment:
Notice that `task_ids` comes from the request body (line 23). Instead of taking the client-provided value, it would be nice to return a response that has task ids that we get from the goal that we fetched from the db on line 21. This way we know for sure that we're returning data from our source of truth (the DB) instead of just echoing back what the user sent to the API (what we get from `request_body["task_ids"]`)

You'd have to grab tasks from goal like:
```py
tasks_from_goal = goal.tasks
task_ids_from_goal = [task.id for task in tasks_from_goal]

 return {
        "id": int(goal_id),
        "task_ids": task_ids_from_goal
}